docs/proposals/25-private-cloud-profiles.md

```diff
@@ -0,0 +1,299 @@
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: privatecloudprofile-modify-special-fields
+rules:
+  - apiGroups: [""]
+    resources: ["privatecloudprofile"]
+    verbs: ["modify-spec-kubernetes", "modify-spec-machineImages"]
+```
+
+### Automated removal of outdated Kubernetes versions and machine image versions
+
+Expired Kubernetes versions in cloud profiles are currently being removed manually. This is not an issue since there is a very limited amount of cloud profiles. However, it any project administrator is allowed to create private cloud profile which can then have its Kubernetes versions expiration dates extended, it cloud become a very cumbersome process to manually remove all the expired Kubernetes versions. It is therefore proposed to implement a custom controller managed my `gardener-controller-manager`, that reconciles private cloud profiles (and possibly even regular cloud profiles) and removes any Kubernetes versions that are past their expiration date. The same could be done for machine images versions.
+
+However, a problem arises as the kubernetes/machine image versions are referenced in the `providerConfig` and would therefore need to be removed manually anyways.
+
+### Cross project sharing
+
+A use case could be defined where a private cloud profile might want to shared across multiple projects and not just be used within the project it was created id. However this use case seems to be very slim so this functionality will not be implemented as of now. Although, when taking a broader view with Private Seeds and Cloud in Country, this feature is going to be necessary at some point. Still, is is not planned to be implemented as of this GEP.
+
+### Multi-inheritance
+
+Theoretically, a private cloud profile could inherit from a cloud profile that already inherits from a cloud profile. However, this should probably not be allowed since it presents some major challenges.
+
+1. The rendering process now has to merge multiple instead of two objects
+2. The parent field becomes more complex because it could point to a cloud profile or to a private cloud profile.
+
+Because of this, multi-inheritance will not be supported as of this GEP.
+
+## Alternatives
+
+### Arbitrary Value Fields
+
+Instead of specifying a private cloud profile resource, an end user could be allowed to enter arbitrary names in fields such as `machineTypes`. The entry would then, if the user enters a wrong value, throw an error when provisioning the resource at the cloud provider. However, this approach does not seam feasible as the metadata that is specified in a cloud profile like amount of cpus and ram is used in the trail clusters to validate quotas. Therefore, an exception would need to be developed for this specific, and possible other, use cases.
```

Comment:
The quantities are used not only for trial clusters (typo, you wrote trail), but also for scale-from-zero if I am not mistaken (cc @rishabh-11).

Reply:
Typo fixed. If you wish, the scale-from-zero use case will be added to clarify the need of the metadata.

Reply:
Added the use case in https://github.com/gardener/gardener/pull/9142/commits/0a315107b5884c4f85da8b0339ada1bd030d3cfc.